Dockerfile

```diff
@@ -1,100 +1,97 @@
 # This dockerfile expects proxies to be set via --build-arg if needed
 # It also expects to be contained in the /navigation2 root folder for file copy
 # Example build command:
-# sudo docker build -t nav2:latest --build-arg http_proxy=http://my.proxy.com:### .
-FROM osrf/ros2:bouncy-desktop
-
-SHELL ["/bin/bash", "-c"]
+# export HTTP_PROXY=http://my.proxy.com:80
+# export CMAKE_BUILD_TYPE=Debug
+# docker build -t nav2:latest --build-arg HTTP_PROXY --build-arg CMAKE_BUILD_TYPE ./
+FROM osrf/ros2:nightly
 
 # setup keys
-# check if proxy is set and get keys, using proxy if it is set
-RUN if [ "$http_proxy" == "" ]; \
+ARG HTTP_PROXY
+RUN if [ "$HTTP_PROXY" != "" ]; \
     then \
-      apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
-      --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
-    else \
       apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
       --keyserver-options http-proxy=$http_proxy \
       --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
     fi
 
-ENV ROS1_DISTRO melodic
-ENV ROS2_DISTRO bouncy
-ENV ROSDISTRO_INDEX_URL 'https://raw.githubusercontent.com/ros/rosdistro/master/index.yaml'
-
-# update latest package versions
-RUN apt-get update
-RUN rm -rf /etc/ros/rosdep/sources.list.d/20-default.list
-
 # install ROS2 dependencies
-RUN apt install -y \
-    build-essential \
-    cmake \
-    git \
-    python3-colcon-common-extensions \
-    python3-pip \
-    python-rosdep \
-    python3-vcstool \
-    wget
+RUN apt-get update && apt-get install -q -y \
+      build-essential \
+      cmake \
+      git \
+      python3-colcon-common-extensions \
+      python3-vcstool \
+      wget \
+    && rm -rf /var/lib/apt/lists/*
 
-# install Fast-RTPS dependencies
-RUN apt install --no-install-recommends -y \
-    libasio-dev \
-    libtinyxml2-dev
-
-# install map_server dependencies
-RUN apt-get install -y \
-    libsdl-image1.2 \
-    libsdl-image1.2-dev \
-    libsdl1.2debian \
-    libsdl1.2-dev
-
-# get the latest nightly ROS2 build -> ros2_ws/ros2_linux
-WORKDIR /ros2_ws
-RUN wget -nv -t 5 https://ci.ros2.org/view/packaging/job/packaging_linux/lastSuccessfulBuild/artifact/ws/ros2-package-linux-x86_64.tar.bz2
-RUN tar -xjf ros2-package-linux-x86_64.tar.bz2
-
-# clone navigation2 repo
-WORKDIR /ros2_ws/navigation2_ws/src
-ARG USER=ros-planning
-RUN git clone https://github.com/$USER/navigation2.git
+# clone ros package repo
+ENV NAV2_WS /opt/nav2_ws
+RUN mkdir -p $NAV2_WS/src
+WORKDIR $NAV2_WS/src
+ARG GIT_REPO_URL=https://github.com/ros-planning/navigation2.git
+RUN git clone $GIT_REPO_URL navigation2
```

Comment:
I sorta feel like we should be `COPY`ing in the source code from the current build context, rather than going through the trouble of cloning from a remote URL, given that much of Travis and CIs sort of deal with merging and staging the repo (build context) before running our scripts. But I guess that depends on what the expected use case of this Dockerfile is. Does one clone this repo to acquire said dockerfile, or expect to build it with the one Dockerfile standalone.

Reply:
This might also make the build more traceable, as there would be so much of a chance for race conditions when building while other commits are merged into the tracked `BRANCH`.

Reply:
There was a discussion about this in #389